### PR TITLE
Group by Sender and Recipient

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Adds ability to group by sender _and_ recipient! Also adds ability to hide them!   

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by michielderoos** [comment]
[Group By Sender and Recipient](https://trello.com/c/GvuRKMuZ/705-group-by-sender-and-recipient)

File: app/client/components/filterModule/FilterModule.jsx [hunk 1]
```diff
@@ -124,15 +138,30 @@ class FilterModule extends React.Component {
...
                 return (
-                  <Option key={key}>
-                    {toTitleCase(replaceUnderscores(key))}
-                  </Option>
+                  <Option key={group}>{allowedGroups[group]["name"]}</Option>
                 );
-              })
-            : null}
+              })}
+            </OptGroup>
```
Comment (tristanhcole) [inline app/client/components/filterModule/FilterModule.jsx:163]: Taking a hint out of @enjeyw OptGroup PR, can we add showing the group name once selected. See example from `filter.jsx`

```
let subList = keys.map(key => {
      //Here we show the label without the group in the dropdown, but with the group once selected
      let label = replaceUnderscores(possibleFilters[key]["name"] || key);
      return (
        <Option key={key} label={label}>
          {label.replace(userGroup, "")}
        </Option>
      );
    });
```

File: app/server/utils/metrics/group.py [hunk 2]
```diff
@@ -49,7 +69,11 @@ def build_query_group_by_with_join(self, query, query_object_model):
                     args = (grouped_query, self.custom_attribute_field_name)
                 else:
                     args = (grouped_query,)
-                return group_joining_strategies[query_object_model.__tablename__][self.group_object_model.__tablename__](*args)
+                if self.sender_or_recipient == 'recipient':
```
Comment (tristanhcole) [inline app/server/utils/metrics/group.py:72]: ` self.sender_or_recipient == 'recipient':` 
sorry coming into this code without context but this seems strange?

**Review by tristanhcole — CHANGES_REQUESTED** [review]
One small change on frontend, otherwise code lgtm and tested.

Don't have enough context/looked into it enough to understand group joining strategies well

File: app/server/utils/metrics/group.py (hunk 2 above)
Comment (michielderoos) [reply app/server/utils/metrics/group.py:72]: Context for this change is [here](https://github.com/teamsempo/SempoBlockchain/pull/558), this is just extending the feature to groups from an existing feature for filters.

File: app/client/components/filterModule/FilterModule.jsx (hunk 1 above)
Comment (michielderoos) [reply app/client/components/filterModule/FilterModule.jsx]: Fixed!

**Review by tristanhcole — APPROVED** [review]
good to go!

**Pull request merged**